frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -66,13 +67,19 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
       resourceFilter,
       dataSelector,
       selectedKey,
+      autoSelect,
     } = nextProps;
 
     if (!loaded) {
       this.setState({ title: <LoadingInline /> });
       return;
     }
-    if (!this.props.loaded || !selectedKey) {
+
+    // If autoSelect is true only then have an item pre-selected based on selectedKey.
+    if (autoSelect) {
+      const selectedItemKey = !selectedKey ? _.get(_.keys(this.state.items), 0) : selectedKey;
+      this.onChange(selectedItemKey);
```

Comment:
`onChange` should only be called if `!selectedKey`, otherwise we are only notifying the parent of something they already know.

Reply:
If you look at onChange it calls this.props.onChange only when the new title is different from what is already present in the state which depends upon the selectedKey, hence we are not notifying the parent unless the selectedKey value changes.

Reply:
It also prevents making unnecessary calls to the parent if the user clicks on a dropdown item more than once.

Reply:
Ah yes this  `this.onChange` and not `this.props.onChange`. That being said, `this.onChange` makes use of `state` and the `state` isn't updated in this function `componentWillReceiveProps` until later in time (async state set). Therefore `this.onChange` actually uses `state` `items` for the name. So IF the items changed during the update, the title could actually end up being a value from the previous items and not the current set.

Reply:
Hmm, but out of all the usecases we have in which one the dropdown items will change ?

Reply:
Nope. Looks like the item key is also the name so it won't cause and issue.